crates/wasmparser/src/validator/operators.rs

```diff
@@ -1163,6 +1186,12 @@ where
             .ok_or_else(|| format_err!(self.offset, "unknown tag {}: tag index out of bounds", at))
     }
 
+    fn global_type_at(&self, at: u32) -> GlobalType {
+        self.resources
+            .global_at(at)
+            .expect("existence should be checked prior to this point")
+    }
```

Comment:
This'll want to return a `Result<_>` to handle out-of-bounds global indices. Also if you're up for it there's probably some other instructions which could use this helper as well, but it's fine to defer that too

Reply:
The other possible users of this were `global.get` and `global.set`, so I refactored them to use the updated version of this function.